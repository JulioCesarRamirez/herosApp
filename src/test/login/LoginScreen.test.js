import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { LoginScreen } from '../../components/login/LoginScreen';
import { types } from '../../types/types';

describe('Testing <LoginScreen/> ', () => {
  const hisotry = {
    replace: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={hisotry} />
    </AuthContext.Provider>
  );

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should do dispatch and navigation', () => {
    const handleClick = wrapper.find('button').prop('onClick');

    handleClick();

    expect(contextValue.dispatch).toHaveBeenLastCalledWith({
      type: types.login,
      payload: {
        name: 'Julio',
      },
    });
    expect(hisotry.replace).toHaveBeenCalledWith('/');
    localStorage.setItem('lastPath', '/dc');
    handleClick();
    expect(hisotry.replace).toHaveBeenCalledWith('/dc');
  });
});
