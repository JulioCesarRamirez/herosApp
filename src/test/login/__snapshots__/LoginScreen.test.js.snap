// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginScreen/>  should render correctly 1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
